src/datagrid/DataGrid.tsx

```diff
@@ -123,10 +125,24 @@ export type DataGridSort = {
     sortFunction: (rows: DataGridRow[], order: SortOrder, columnName: string) => Promise<DataGridRow[]>;
 };
 
-export type DatagridData = {
-    rows: DataGridRow[];
-    columns?: DataGridColumn[];
-    footer?: DataGridFooter;
+/**
+ * Props for DataGridPagination :
+ * @param {className} CSS
+ * @param {style} CSS styles
+ * @param {currentPage} Index of the currently displayed page, starting from 1
+ * @param {pageSize} Number of items displayed per page. Defaults to 10
+ * @param {totalItems} Total number of items present in the datagrid, after the filters have been applied
+ * @param {lastPage} Index of the last page for the current data
+ * @param {getPage} custom function to get page data for given page number
+ */
+type DataGridPaginationProps = {
+    className?: string;
+    style?: any;
+    currentPage?: number;
+    pageSize?: number;
+    pageSizes?: number[];
+    totalItems: number;
+    getPageData?: (pageIndex: number, pageSize: number) => Promise<DataGridRow[]>;
```

Comment:
Should this be populatePageData or loadPageData?

Reply:
datagrid is getting data from outside so getPageData seems logical.